### PR TITLE
Modify existing example of adding checker

### PR DESCRIPTION
**Pull request by xennygrimmato**
The example of [adding-checker](http://fbinfer.com/docs/adding-checkers.html) uses an old revision for explanation. I have used a function defined in `patternMatch.ml` to replicate the use using the latest revision.

### CODE REVIEW

File: _docs/04-adding-checkers.md [hunk 1]
```diff
@@ -89,63 +89,52 @@ let callback_my_simple_checker { Callbacks.proc_desc; proc_name } =
...
-		(js.method_name = "println") && (class_name = "PrintStream")
-	| _ -> false
-```
-
-And in file ```procname.mli``` add the respective signature:
-
-```ocaml
-(** [java_is_print_method pn] returns true if [pn] is an output method **)
-val java_is_print_method : t -> bool
-```
```
Comment (jvillard) [inline _docs/04-adding-checkers.md]: why did you remove these paragraphs?

Comment (xennygrimmato) [inline _docs/04-adding-checkers.md]: I removed the paragraphs because I wrote the `is_println` expression in `checkers.ml` (L117-L122).
Maybe, I should move that to `procname.ml` and retain the content?

File: _docs/04-adding-checkers.md [hunk 2]
```diff
@@ -89,63 +89,52 @@ let callback_my_simple_checker { Callbacks.proc_desc; proc_name } =
   Cfg.Procdesc.iter_instrs do_instr proc_desc
 
 ```
+The `checkers/PatternMatch.ml` module contains the `java_proc_name_with_class_method` function which we can use for matching the required pattern.
 
-Each node is represented using the type ```instr``` from the Smallfoot Intermediate Language (SIL). Take a look at ```sil.mli``` to get familiar with all the types. All source code languages supported by Infer are converted to this representation.
+Each node is represented using the type ```instr``` from the Smallfoot Intermediate Language (SIL). Take a look at ```IR/sil.rei``` to get familiar with all the types. All source code languages supported by Infer are converted to this representation.
```
Comment (jvillard) [inline _docs/04-adding-checkers.md]: it's IR/Sil.rei (capital S)

File: _docs/04-adding-checkers.md [hunk 3]
```diff
@@ -89,63 +89,52 @@ let callback_my_simple_checker { Callbacks.proc_desc; proc_name } =
...
-	Sil.Const (Sil.Cfun name_of_function),
+	Sil.Const (Const.Cfun name_of_function),
 	list_of_arguments,
 	location,
 	call_flags
 )
 ```
 
-I hope this looks straight forward. Argument ```call_flags``` holds information about the function, such as whether it is virtual or not. Again, this is specified in the file ```sil.mli```.
+I hope this looks straight forward. Argument ```call_flags``` holds information about the function, such as whether it is virtual or not. Again, this is specified in the file ```sil.rei```.
```
Comment (jvillard) [inline _docs/04-adding-checkers.md]: Sil.rei

File: _docs/04-adding-checkers.md (hunk 1 above)
Comment (jvillard) [inline _docs/04-adding-checkers.md]: I think it's fine as you wrote it. Even though it was nice to have a little tutorial on where to put such utility functions, it's not strictly related to this guide.

**Comment by xennygrimmato** [comment]
I've changed `sil.rei` to `Sil.rei`.

**Comment by jvillard** [comment]
Thanks @xennygrimmato!
